eth/stagedsync/stage_txlookup.go

```diff
@@ -226,11 +264,30 @@ func deleteTxLookupRange(tx kv.RwTx, logPrefix string, blockFrom, blockTo uint64
 				return err
 			}
 		}
-		if cfg.isBor {
-			borTxHash := types.ComputeBorTxHash(blocknum, blockHash)
-			if err := tx.Delete(kv.BorTxLookup, borTxHash.Bytes()); err != nil {
-				return err
-			}
+
+		return nil
+	}, etl.IdentityLoadFunc, etl.TransformArgs{
+		Quit:            ctx.Done(),
+		ExtractStartKey: dbutils.EncodeBlockNumber(blockFrom),
+		ExtractEndKey:   dbutils.EncodeBlockNumber(blockTo),
+		LogDetailsExtract: func(k, v []byte) (additionalLogArguments []interface{}) {
+			return []interface{}{"block", binary.BigEndian.Uint64(k)}
+		},
+	})
+}
+
+// deleteTxLookupRange - [blockFrom, blockTo)
+func deleteBorTxLookupRange(tx kv.RwTx, logPrefix string, blockFrom, blockTo uint64, ctx context.Context, cfg TxLookupCfg) error {
+	return etl.Transform(logPrefix, tx, kv.HeaderCanonical, kv.BorTxLookup, cfg.tmpdir, func(k, v []byte, next etl.ExtractNextFunc) error {
+		blocknum, blockHash := binary.BigEndian.Uint64(k), common.CastToHash(v)
+
+		if !rawdb.HasBorReceipts(tx, blocknum) {
```

Comment:
i think our delete will not return error if key doesn't exists

Reply:
It would be slower as well to actually check for receipt than calculating the bortxhash I believe.